src/commands/subdomain.rs

```diff
@@ -126,6 +136,44 @@ pub fn subdomain(name: &str, user: &GlobalUser, target: &Target) -> Result<(), f
     Ok(())
 }
 
+pub fn set_subdomain(name: &str, user: &GlobalUser, target: &Target) -> Result<(), failure::Error> {
+    if target.account_id.is_empty() {
+        failure::bail!(format!(
+            "{} You must provide an account_id in your wrangler.toml before creating a subdomain!",
+            emoji::WARN
+        ))
+    }
+    let subdomain = Subdomain::get_as_option(&target.account_id, user);
+    match subdomain {
+        Ok(None) => register_subdomain(&name, &user, &target),
+        Ok(Some(subdomain)) => {
+            let msg = format!("This account already has a registered subdomain. You can only register one subdomain per account. Your subdomain is {}.workers.dev", subdomain);
+            message::user_error(&msg);
+            Ok(())
+        }
+        Err(error) => Err(error),
+    }
+}
+
+pub fn get_subdomain(user: &GlobalUser, target: &Target) -> Result<(), failure::Error> {
+    let subdomain = Subdomain::get_as_option(&target.account_id, user);
```

Comment:
You could simplify this with a `?` at the end of the `let subdomain` line and then the `match` statement could just be `Some(subdomain) => ` and `None =>`

Reply:
Ah right – I still haven't internalized all of the useful syntax and idioms. Fixed in https://github.com/cloudflare/wrangler/pull/747/commits/1dd74e1274363d50167619f29efc1cf51938f185.